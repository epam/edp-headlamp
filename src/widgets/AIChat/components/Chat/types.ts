import { CodemieSecretData, ConversationItem } from '../../types';

export interface ChatProps {
  codemieSecretData: CodemieSecretData;
  conversation: ConversationItem;
  updateConversation: (conversation: ConversationItem) => void;
  requestError: Error;
}
