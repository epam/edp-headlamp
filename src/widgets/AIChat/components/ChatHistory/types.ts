import { ConversationItem, ConversationState } from '../../types';

export interface ChatHistoryProps {
  conversations: ConversationItem[];
  activeConversation: ConversationItem | null;
  setActiveConversation: React.Dispatch<React.SetStateAction<ConversationItem | null>>;
  setConversations: React.Dispatch<React.SetStateAction<ConversationState>>;
  createNewConversation: () => void;
}
