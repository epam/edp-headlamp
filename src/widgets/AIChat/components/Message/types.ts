import React from 'react';
import { ChatEntity } from '../../types';

export interface MessageProps {
  entityRole: ChatEntity;
  createdAt: string;
  content: React.ReactNode | string;
}
