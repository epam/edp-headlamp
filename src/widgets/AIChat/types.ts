import { ValueOf } from '../../types/global';
import { CHAT_ENTITY } from './constants';

export type ChatEntity = ValueOf<typeof CHAT_ENTITY>;

export interface ChatItemRequest {
  message: string;
}

export interface ProcessedChunk {
  id: string;
  text: string;
  isCode: boolean;
}

export interface ChatItemResponse {
  message: string | undefined; // is available on streaming finish
  processedChunks: ProcessedChunk[];
}

export interface HistoryStateItem {
  id: string;
  createdAt: string;
  request: ChatItemRequest;
  response?: ChatItemResponse;
}

export interface HistoryPayloadItem {
  createdAt: string;
  message: string;
  role: ChatEntity;
}

export interface ConversationItem {
  conversationId: string;
  conversationHistory: HistoryStateItem[];
  assistantID: string;
  assistantName: string;
  assistantContext: string;
  prompt: string;
  llmModel: string;
  assistantMode: boolean;
  file: File | null;
  stream: boolean;
  _debug: boolean;
  topK: number;
  codeMode?: boolean;
  kbMode?: boolean;
  kbType?: string;
  kbName?: string;
  appName?: string;
  repoName?: string;
  indexType?: string;
}

export interface ConversationPayload {
  conversationId: string;
  text: string;
  file: File | null;
  prompt: string;
  history: HistoryPayloadItem[];
  llmModel: string;
  stream: boolean;
  _debug: boolean;
  topK: number;
  code_fields?: {
    appName: string;
    repoName: string;
    indexType: string;
  };
}

export interface ConversationResponseChunk {
  time_elapsed: number;
  input_tokens_used: number;
  tokens_used: number;
  generated_chunk: string;
  generated: string;
  thought: string;
  last: boolean;
  debug: null;
}

export interface ConversationState {
  [key: string]: ConversationItem;
}

export interface CodemieSecretData {
  apiUrl: string;
  assistantId: string;
  token: string;
}
