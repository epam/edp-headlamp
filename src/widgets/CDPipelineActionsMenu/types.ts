import { ACTION_MENU_TYPES } from '../../constants/actionMenuTypes';
import { CDPipelineKubeObjectInterface } from '../../k8s/groups/EDP/CDPipeline/types';
import { ValueOf } from '../../types/global';
import { PermissionSet } from '../../types/permissions';

export interface CDPipelineActionsMenuProps {
  data: {
    CDPipelineData: CDPipelineKubeObjectInterface;
  };
  backRoute?: string;
  variant?: ValueOf<typeof ACTION_MENU_TYPES>;
  anchorEl?: HTMLElement;
  handleCloseResourceActionListMenu?: () => void;
  permissions: PermissionSet;
}
