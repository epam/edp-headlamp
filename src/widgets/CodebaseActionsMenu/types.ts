import { ACTION_MENU_TYPES } from '../../constants/actionMenuTypes';
import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';
import { ValueOf } from '../../types/global';
import { PermissionSet } from '../../types/permissions';

export interface CodebaseActionsMenuProps {
  data: {
    codebaseData: EDPCodebaseKubeObjectInterface;
  };
  backRoute?: string;
  variant?: ValueOf<typeof ACTION_MENU_TYPES>;
  anchorEl?: HTMLElement;
  handleCloseResourceActionListMenu?: () => void;
  permissions: PermissionSet;
}
