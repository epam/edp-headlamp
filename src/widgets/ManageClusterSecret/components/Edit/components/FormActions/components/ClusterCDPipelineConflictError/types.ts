import { StageKubeObjectInterface } from '../../../../../../../../k8s/groups/EDP/Stage/types';

export interface ClusterCDPipelineConflictErrorProps {
  conflictedStage: StageKubeObjectInterface;
  clusterName: string;
}
