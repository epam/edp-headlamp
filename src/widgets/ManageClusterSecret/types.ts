import { SecretKubeObjectInterface } from '../../k8s/groups/default/Secret/types';
import { PermissionsConfig } from '../../providers/Permissions/types';
import { FORM_MODES } from '../../types/forms';
import { ValueOf } from '../../types/global';
import { widgetPermissionsToCheck } from './constants';
import { CLUSTER_FORM_NAMES } from './names';

export type WidgetPermissions = PermissionsConfig<typeof widgetPermissionsToCheck>;

export interface ManageClusterSecretDataContext {
  handleClosePlaceholder?: () => void;
  mode: ValueOf<typeof FORM_MODES>;
  permissions: WidgetPermissions;
  currentElement?: SecretKubeObjectInterface;
  ownerReference: string | undefined;
}

export interface ManageClusterSecretProps {
  formData: ManageClusterSecretDataContext;
}

export type ManageClusterSecretValues = {
  [key in ValueOf<typeof CLUSTER_FORM_NAMES>]: any;
};
