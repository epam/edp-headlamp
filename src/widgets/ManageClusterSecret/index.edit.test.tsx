/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { FORM_MODES } from '../../types/forms';
import { ManageClusterSecret } from './index';

test('renders ManageClusterSecret Edit component', () => {
  render(
    <TestWrapper>
      <ManageClusterSecret
        formData={{
          currentElement: {
            apiVersion: 'v1',
            kind: 'Secret',
            metadata: {
              name: 'test-cluster-name',
              labels: { 'argocd.argoproj.io/secret-type': 'cluster' },
              creationTimestamp: '',
              uid: '',
            },
            data: {
              config:
                'eyJhcGlWZXJzaW9uIjoidjEiLCJraW5kIjoiQ29uZmlnIiwiY3VycmVudC1jb250ZXh0IjoiZGVmYXVsdC1jb250ZXh0IiwicHJlZmVyZW5jZXMiOnt9LCJjbHVzdGVycyI6W3siY2x1c3RlciI6eyJzZXJ2ZXIiOiJ0ZXN0LWNsdXN0ZXItaG9zdCJ9LCJuYW1lIjoiZGVmYXVsdC1jbHVzdGVyIn1dLCJjb250ZXh0cyI6W3siY29udGV4dCI6eyJjbHVzdGVyIjoiZGVmYXVsdC1jbHVzdGVyIiwidXNlciI6ImRlZmF1bHQtdXNlciJ9LCJuYW1lIjoiZGVmYXVsdC1jb250ZXh0In1dLCJ1c2VycyI6W3sidXNlciI6eyJ0b2tlbiI6InRlc3QtY2x1c3Rlci10b2tlbiJ9LCJuYW1lIjoiZGVmYXVsdC11c2VyIn1dfQ==',
            },
          },
          mode: FORM_MODES.EDIT,
          permissions: {
            create: {
              Secret: {
                allowed: true,
              },
            },
            update: {
              Secret: {
                allowed: true,
              },
            },
            delete: {
              Secret: {
                allowed: true,
              },
            },
          },
        }}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('form');
  expect(dialog).toMatchSnapshot();
});
