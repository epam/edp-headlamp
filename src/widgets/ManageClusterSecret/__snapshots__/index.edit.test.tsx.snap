// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageClusterSecret Edit component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-2"
                    id="mui-2-label"
                  >
                    Cluster Name
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled=""
                      id="mui-2"
                      name="clusterName"
                      placeholder="Enter cluster name"
                      type="text"
                      value="test-cluster-name"
                    />
                    <span />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-14lo706"
                      >
                        <span>
                          Cluster Name
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-4"
                    id="mui-4-label"
                  >
                    Cluster Host
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      id="mui-4"
                      name="clusterHost"
                      placeholder="Enter cluster host"
                      type="text"
                      value="test-cluster-host"
                    />
                    <span />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-14lo706"
                      >
                        <span>
                          Cluster Host
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-6"
                    id="mui-6-label"
                  >
                    Cluster Token
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      id="mui-6"
                      name="clusterToken"
                      placeholder="Enter cluster token"
                      type="password"
                      value="test-cluster-token"
                    />
                    <div
                      class="MuiStack-root css-niqf4j-MuiStack-root"
                    >
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <span />
                      </div>
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="0"
                          type="button"
                        >
                          <span />
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-14lo706"
                      >
                        <span>
                          Cluster Token
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-qh9fv7-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-jhdc2n-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <div
                  class="MuiStack-root css-jfdv4h-MuiStack-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      style="margin: 0px;"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1pkws1h-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="skipTLSVerify"
                          type="checkbox"
                          value="false"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1ezega9-MuiTypography-root"
                      >
                        <span
                          style="display: inline-block; margin-top: 0.125rem;"
                        >
                          <span
                            class="makeStyles-labelWrap-3"
                          >
                            <span
                              class="MuiTypography-root MuiTypography-body1 makeStyles-label-4 css-1ezega9-MuiTypography-root"
                            >
                              Skip TLS verification
                            </span>
                            <div
                              class="MuiBox-root css-0"
                              style="height: 1.25rem;"
                            />
                          </span>
                        </span>
                      </span>
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
              >
                <div
                  class="MuiStack-root css-jfdv4h-MuiStack-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-8dtgre-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="false"
                        for="mui-8"
                        id="mui-8-label"
                      >
                        Cluster Certificate
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                          id="mui-8"
                          name="clusterCertificate"
                          placeholder="Enter cluster certificate"
                          type="text"
                          value=""
                        />
                        <span />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-yjsfm1"
                          >
                            <span>
                              Cluster Certificate
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1xu1mzw-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1llbb2a-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  undo changes
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-vw8gqz-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  save
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
